lib/adapters/document-sync-adapter.ts

```diff
@@ -166,6 +167,7 @@ export class TextEditorSyncAdapter {
 
     this._disposable.add(
       editor.getBuffer().onWillSave(this.willSave.bind(this)),
+      editor.getBuffer().onWillSave(this.willSaveWaitUntil.bind(this)),
```

Comment:
This is a good suggestion 👍 

> You could change the `documentSyncKind` parameter to be of type `TextDocumentSyncOptions | TextDocumentSyncKind`

I changed it to just `TextDocumentSyncOptions` for consistency. See d54dae1 & 4f7b9aa.

I would go further with this and fix `didSave`/`didClose`/`didOpen`/etc. in the same way. Currently if server sends `textDocumentSync` capabilities with `openClose: false`, atom will still send open/close notifications. It's not a big deal, but it's easy to fix now. I can open another PR if you support this suggestion.

Reply:
If you have the time and it's easy, I'd say go for it.  You could just do it as a separate commit in this PR instead of a whole new PR.  Thanks again!

Reply:
OK. If this is ready, let's merge it and I'll submit another PR when I have time. I would also like it to be separate as these changes are independent.